### PR TITLE
chore: auto generate esbuild test scripts.

### LINKED ISSUE #78: Auto generated esbuild tests.
1. A tool automatically converts esbuild tests to snapshot test

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
### Description
1. the auto scripts look basically available, I regenerated the tests in `default` dir, the only difference is the entry name, but there are still some issues that need to be addressed.
-  Not all esbuild inline snapshots are coded as literal. e.g., the` files`  field referenced a top-level binding.
image
But luckily, only a few tests are written like this.

We could use this for now, and improve in the future.
2. Closed https://github.com/rolldown-rs/rolldown/issues/78

### Test Plan

---

### CODE REVIEW

**Comment by hyf0** [comment]
We might need to config `test.config.json` to change the behaviors of rolldown to make the test passed. So `test.config.json` should only be created while there isn't one.

I'm not sure if we need to reference esbuild tests using git submodules, let's have a discussion of it.

Above all, well done👍

**Comment by Boshen** [comment]
Nice job!

> I'm not sure if we need to reference esbuild tests using git submodules, let's have a discussion of it.

I don't think submodule is required because this is based on a script, let's just run it periodically.

File: yarn.lock
Comment (Boshen) [inline yarn.lock:1]: Why are we using yarn?

**Review by hyf0 — APPROVED** [review]

File: scripts/bundler_default.go
Comment (Boshen) [inline scripts/bundler_default.go:1]: This is huge! shall we just download this from github? https://raw.githubusercontent.com/evanw/esbuild/main/internal/bundler_tests/bundler_default_test.go

File: scripts/genESbuildTest.js
Comment (Boshen) [inline scripts/genESbuildTest.js:1]: Can you add a top level documentation on what this does, and also how to run this?

File: yarn.lock
Comment (hyf0) [reply yarn.lock:1]: yarn3 could works well with docker while pnpm doesnt for publishing packages. This is the experience from @Brooooooklyn.

File: scripts/genESbuildTest.js
Comment (IWANABETHATGUY) [reply scripts/genESbuildTest.js:1]: Sure

File: scripts/bundler_default.go
Comment (IWANABETHATGUY) [reply scripts/bundler_default.go:1]: yes,  I would remove this golang file, after I add some comments about how to use this scripts.

File: scripts/genESbuildTest.js
Comment (IWANABETHATGUY) [reply scripts/genESbuildTest.js:1]: Done

**Comment by Boshen** [comment]
You may merge if CI is running these tests.

**Review by underfin — APPROVED** [review]

**Pull request merged**